client/src/components/RichTextEditor.js

```diff
@@ -1,232 +1,258 @@
-      "M256 1012c-65.176 0-126.45-25.38-172.534-71.464-95.134-95.136-95.134-249.934 0-345.070l87.764-87.764c20.308-20.306 53.234-20.306 73.54 0 20.308 20.306 20.308 53.232 0 73.54l-87.764 87.764c-54.586 54.586-54.586 143.406 0 197.992 26.44 26.44 61.598 41.002 98.994 41.002s72.552-14.562 98.998-41.006l192-191.998c54.584-54.586 54.584-143.406 0-197.992-20.308-20.308-20.306-53.232 0-73.54 20.306-20.306 53.232-20.306 73.54 0.002 95.132 95.134 95.132 249.932 0.002 345.068l-192.002 192c-46.090 46.088-107.364 71.466-172.538 71.466z"
-    ],
-    // React component providing the UI to manage entities of this type.
-    source: LinkSourceAnet,
-    // React component to display inline entities.
-    decorator: LinkAnet,
-    // React component to display block-level entities.
-    block: PropTypes.func,
-    // Array of attributes the entity uses, to preserve when filtering entities on paste. If undefined, all entity data is preserved.
-    attributes: null,
-    // Attribute - regex mapping, to whitelist entities based on their data on paste.
-    // For example, { url: '^https:' } will only preserve links that point to HTTPS URLs.
-    whitelist: {
-      href: "."
-    }
-  }
-}
+import React, { useCallback, useMemo, useState } from "react"
+import { createEditor, Text } from "slate"
+import { withHistory } from "slate-history"
+import { jsx } from "slate-hyperscript"
+import {
+  Editable,
+  Slate,
+  useFocused,
+  useSelected,
+  withReact
+} from "slate-react"
+import { getUrlFromEntityInfo } from "utils_links"
 
-const importerConfig = {
-  htmlToEntity: (nodeName, node, createEntity) => {
-    if (nodeName === "a") {
-      return createEntity(ENTITY_TYPE.LINK, "IMMUTABLE", {
-        url: node.href
-      })
-    }
+const RichTextEditor = ({ value, onChange, onHandleBlur, className }) => {
+  const withAnetLink = editor => {
+    const { isVoid, isInline } = editor
 
-    if (nodeName === "hr") {
-      return createEntity(ENTITY_TYPE.HORIZONTAL_RULE, "IMMUTABLE", {})
-    }
+    editor.isVoid = element =>
+      element.type === "anet-link" ? true : isVoid(element)
 
-    return null
-  },
-  htmlToBlock: nodeName => {
-    if (nodeName === "hr" || nodeName === "img") {
-      // "atomic" blocks is how Draft.js structures block-level entities.
-      return "atomic"
-    }
+    editor.isInline = element =>
+      element.type === "anet-link" ? true : isInline(element)
 
-    return null
-  },
-  htmlToStyle: (nodeName, node, currentStyle) => {
-    return currentStyle
+    return editor
   }
-}
+  const editor = useMemo(
+    () => withReact(withHistory(withAnetLink(createEditor()))),
+    []
+  )
+  const document = new DOMParser().parseFromString(value || "", "text/html")
+  const deserialized = deserialize(document.body)
```

Comment:
If this is expensive, it could perhaps be memoized:
```
  const deserialized = useMemo(() => {
    const document = new DOMParser().parseFromString(value || "", "text/html")
    return deserialize(document.body)
  }, [value])
```